airflow/utils/task_group.py

```diff
@@ -135,8 +147,9 @@ def __init__(
         # so that we can optimize the number of edges when entire TaskGroups depend on each other.
         self.upstream_group_ids: Set[Optional[str]] = set()
         self.downstream_group_ids: Set[Optional[str]] = set()
-        self.upstream_task_ids: Set[Optional[str]] = set()
-        self.downstream_task_ids: Set[Optional[str]] = set()
+        # Since the parent class defines these as read-only properties, we can 't just do `self.x = ...`
+        self.__dict__['upstream_task_ids'] = set()
+        self.__dict__['downstream_task_ids'] = set()
```

Comment:
Can we not change the parent class definition to be plain class-level variable declarations?

Reply:
No, because on BaseOperator these are defined as properties.